src/Evolution/DgSubcell/Actions/Initialize.hpp

```diff
@@ -0,0 +1,162 @@
+ * mesh, the initial RDMP parameters \f$\delta_0\f$ and \f$\epsilon\f$, and the
+ * Persson TCI parameter \f$\alpha\f$. The `apply` function must return a `bool`
+ * that is `true` if the cell is troubled. If the cell is troubled then
+ * `Tags::ActiveGrid` is set to `subcell::ActiveGrid::Subcell`, the
+ * `System::variables_tag` become the variables on the subcell grid set by call
+ * into `evolution::Initialization::Actions::SetVariables`, and
+ * `Tags::Inactive<System::variables_tag>` become the (inadmissible) DG
+ * solution.
+ *
+ * GlobalCache:
+ * - Uses:
+ *   - `subcell::Tags::SubcellOptions`
+ * DataBox:
+ * - Uses:
+ *   - `domain::Tags::Mesh<Dim>`
+ *   - `domain::Tags::Element<Dim>`
+ *   - `System::variables_tag`
+ * - Adds:
+ *   - `subcell::Tags::Mesh<Dim>`
+ *   - `subcell::Tags::ActiveGrid`
+ *   - `subcell::Tags::DidRollback`
+ *   - `subcell::Tags::Inactive<System::variables_tag>`
+ *   - `subcell::Tags::TciGridHistory`
+ *   - `subcell::Tags::NeighborDataForReconstructionAndRdmpTci<Dim>`
+ *   - `subcell::fd::Tags::InverseJacobianLogicalToGrid<Dim>`
+ *   - `subcell::fd::Tags::DetInverseJacobianLogicalToGrid`
+ *   - `subcell::Tags::LogicalCoordinates<Dim>`
+ *   - `subcell::Tags::Corodinates<Dim, Frame::Grid>` (as compute tag)
+ *   - `subcell::Tags::TciStatusCompute<Dim>`
+ * - Removes: nothing
+ * - Modifies:
+ *   - `System::variables_tag` if the cell is troubled
+ */
+template <typename Metavariables>
+struct Initialize {
+  using const_global_cache_tags = tmpl::list<Tags::SubcellOptions>;
+
+  using simple_tags = tmpl::list<
+      Tags::Mesh<Metavariables::volume_dim>, Tags::ActiveGrid,
+      Tags::DidRollback,
+      Tags::Inactive<typename Metavariables::system::variables_tag>,
+      Tags::TciGridHistory,
+      Tags::NeighborDataForReconstructionAndRdmpTci<Metavariables::volume_dim>,
+      fd::Tags::InverseJacobianLogicalToGrid<Metavariables::volume_dim>,
+      fd::Tags::DetInverseJacobianLogicalToGrid>;
+  using compute_tags = tmpl::list<
+      Tags::LogicalCoordinatesCompute<Metavariables::volume_dim>,
+      ::domain::Tags::MappedCoordinates<
+          ::domain::Tags::ElementMap<Metavariables::volume_dim, Frame::Grid>,
+          subcell::Tags::Coordinates<Metavariables::volume_dim, Frame::Logical>,
+          subcell::Tags::Coordinates>,
+      Tags::TciStatusCompute<Metavariables::volume_dim>>;
+
+  template <typename DbTagsList, typename... InboxTags, typename ArrayIndex,
+            typename ActionList, typename ParallelComponent>
+  static std::tuple<db::DataBox<DbTagsList>&&> apply(
+      db::DataBox<DbTagsList>& box,
+      const tuples::TaggedTuple<InboxTags...>& inboxes,
+      const Parallel::GlobalCache<Metavariables>& cache,
+      const ArrayIndex& array_index, ActionList /*meta*/,
+      const ParallelComponent* const /*meta*/) noexcept {
+    const SubcellOptions& subcell_options = db::get<Tags::SubcellOptions>(box);
+    // Intentional copy of mesh
```

Comment:
Me either, I'll switch to a reference.  I think at some point I was considering changing `domain::Tags::Mesh<Dim>` when switching from DG to subcell because compute tags make life terrible, but I decided against that design because it just makes things very confusing. Maybe it'll be possible once everything is in and working :shrug: